gems/sqf-aspects/fn_addFunctionAspects.sqf

```diff
@@ -0,0 +1,106 @@
+#include "script_component.hpp"
+/*
+ * Author: xetra11
+ *
+ * This function will enrich existing functions with aspects before and after the function call.
+ * This way it will remove boilerplate code mostly used for debugging purpose.
+ * Aspects add this functionality while making it more difficult to maintain the code base.
+ * Only use if really needed!
+ *
+ * Arguments:
+ * 0: _beforeAspect <STRING | CODE> - before aspect code as string
+ * 1: _afterAspect <STRING | CODE> - after aspect code as string
+ * 2: _functionTargets <ARRAY> - [<enrichmentMode> (, <functionName> | <tagPrefix> )]
+
+ * enrichmentMode : 0 = all functions that start with <tagPrefix>
+ * enrichmentMode : 1 = only one function as <functionName>
+ *
+ * Return Value:
+ *
+ * Example:
+ *
+ *     [{ diag_log ">> __FUNC_NAME__"},
+ *      {diag_log "<< __FUNC_NAME__"},
+ *      [0,"myTag"]] call coopr_fnc_addFunctionAspects;
+ *
+ *     [{ diag_log "before func"},
+ *      {diag_log "after func"},
+ *      [1,"myTag_fnc_myFooFunc"]] call coopr_fnc_addFunctionAspects;
+ *
+ * Public: No
+ *
+ * Scope: Global
+ */
+
+#define FUNCS_BY_TAG 0
+#define FUNC_NAME 1
+
+params [["_beforeAspect", {}],
+        ["_afterAspect", {}],
+        ["_enrichmentMode", [], [[0, ""]]]];
+
+if (isNil "IGNORE_FUNCTIONS") then {
+    IGNORE_FUNCTIONS = [];
```

Comment:
all uppercase names usually signify a macro.
This is just a global variable without tags, which makes it even worse.

Reply:
so it should be `gemsIgnoreFunctions` ?

Reply:
Its usually "TAG_variableName"